tests/codegen/call_offsetof.cpp

```diff
@@ -0,0 +1,23 @@
+#include "common.h"
+
+namespace bpftrace {
+namespace test {
+namespace codegen {
+
+TEST(codegen, call_offsetof)
+{
+  test("struct Foo { int x; long l; char c; } BEGIN { @x = offsetof(struct Foo, x); @c = offsetof(struct Foo, c); }",
+
+       NAME);
+}
+
+TEST(codegen, call_offsetof_task_struct)
+{
+  test("#include <linux/sched.h> \
+  BEGIN { @x = offsetof(struct Foo, comm); }",
```

Comment:
I'm afraid that this will be very flaky - a change of `struct task_struct` layout in kernel will break it. I don't think that we need this test.

Reply:
instead a test for the more exotic structs might be good. What about function pointers, anonymous structs and other nesting?

Reply:
Add `union`, `function pointers`, `anonymous structs` unit test in `tests/semantic_analyser.cpp`, looks like works:

```
$ sudo ./bpftrace_test --gtest_filter=semantic_analyser.call_offsetof
Note: Google Test filter = semantic_analyser.call_offsetof
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from semantic_analyser
[ RUN      ] semantic_analyser.call_offsetof
[       OK ] semantic_analyser.call_offsetof (1594 ms)
[----------] 1 test from semantic_analyser (1594 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (1594 ms total)
[  PASSED  ] 1 test.
```